runtests.py

```diff
@@ -346,6 +346,39 @@ def update_cpp17_extension(ext):
 
     return EXCLUDE_EXT
 
+def update_cpp20_extension(ext):
```

Comment:
I think we have enough copies of this function by now to merit a configurable version, i.e. have one function, pass in all relevant compiler versions, and let it return a closure function that configures the extension build. Then call the outer function three times in `EXT_EXTRAS`, for C++11/17/20, and then some more in the future.

Reply:
I refactored the setup.